tardis/model/base.py

```diff
@@ -285,38 +381,23 @@ def velocity(self):
 
     @property
     def v_inner(self):
-        return self.velocity[:-1]
+        return self.model_state.v_inner
 
     @property
     def v_outer(self):
-        return self.velocity[1:]
+        return self.model_state.v_outer
 
     @property
     def v_middle(self):
         return 0.5 * self.v_inner + 0.5 * self.v_outer
 
     @property
     def density(self):
-        density = (
-            self.homologous_density.calculate_density_at_time_of_simulation(
-                self.time_explosion
-            )
-        )
-        return density[
-            self.v_boundary_inner_index + 1 : self.v_boundary_outer_index + 1
-        ]
+        return self.model_state.density
 
     @property
     def abundance(self):
-        if not self.raw_isotope_abundance.empty:
-            self._abundance = self.raw_isotope_abundance.decay(
-                self.time_explosion
-            ).merge(self.raw_abundance)
-        abundance = self._abundance.loc[
-            :, self.v_boundary_inner_index : self.v_boundary_outer_index - 1
-        ]
-        abundance.columns = range(len(abundance.columns))
-        return abundance
+        return self.model_state.abundance
```

Comment:
https://github.com/tardis-sn/tardis/blob/bf9b70833cfd260c651a1d52e83c37125f4ca52b/tardis/model/tests/test_base.py#L286-L288

This test failed in the last commit because `abundance` depends upon `raw_isotope_abundance`, and changing the latter does not change the former. 
One way to do this is creating another function for calculating abundance, and assigning the result to `model_state.abundance`. This is done in https://github.com/tardis-sn/tardis/pull/1870/commits/1254a1e7b6e6c4bf349a041d604fb30d72dfd3e5 , is it a good way to do this?

Reply:
so - how about you get isotope abundances and then internally we calculate elemental abundances

Reply:
Well, we would also need to pass in `raw_abundance`, `v_boundary_inner_index` and `v_boundary_outer_index` too, because the abundance calculation depends on these things. Would that be alright?

Reply:
no - we need to cut the isotope abundance beforehand.

Reply:
we should never use `v_boundary_inner_index` in any of the model state

Reply:
a good way to think about it - the modelstate essentially only stores. There are no cutting operations within the classs.